xblock/test/test_core.py

```diff
@@ -435,8 +435,9 @@ class FieldTester(XBlock):
 
 def test_object_identity():
     # Check that values that are modified are what is returned
-    class FieldTester(ScopedStorageMixin):
-        """Toy class for ModelMetaclass and field access testing"""
+    @Blocklike.needs("field-data")
+    class FieldTester(Blocklike):
```

Comment:
RuntimeServicesMixin defined the `@needs` decorator, which allowed its subclass ScopedStorageMixin to invoke `@needs("field-data")`.

However, now Blocklike defines `@needs`, so Blocklike itself cannot invoke `@needs("field-data")`.

Thus, XBlock and XBlockAside now each invoke `@needs("field-data")` individually, and thus tests that subclass Blocklike must also invoke `@needs("field-data")`.

Reply:
Can you manually invoke the decorator on `Blocklike` after it's defined? I haven't done something like this in ages, but I think it'd be something like this?

```python
class Blocklike:
    # lots of stuff here

Blocklike = Blocklike.needs(Blocklike, "field-data")
```